design/004-native-rstudio-sessions/004-native-rstudio-sessions.md

```diff
@@ -0,0 +1,198 @@
+    A2[Read fewer hardcoded Jinja templates]
+    B[Parse variables from k8s manifest]
+    C[Parse patches from k8s manifest]
+    C1[Read Jinja templates based on session type]
+    D[Render Jinja tempaltes to yaml]
+    E[Patch yaml] 
+    F[Create k8s resources]
+    
+    A1 --> B
+    A1 --> C
+    A1 --> C1
+    B --> D
+    C1 --> D
+    A2 --> D
+    D --> E
+    C --> E
+    E --> F
+```
+
+### Current assumptions hardcoded in Amalthea Jinja templates
+- Session is a statefulset, contains session container and authentication proxy
+- Session container listens on port 8888
+- Authentication proxy listens on port 4180
+- Service points to authentication proxy (4180)
+- Ingress points to service which leads only to authentication proxy
+- Secrets for Jupyterlab and authentication
+- Configmap for Jupyterlab config (mounted in session container)
+- PVC for session storage
+
+### Some of these assumptions will remain hardcoded
+- PVC/Storage, but expect the user to patch in volume mount
+- Ingress, keep as is point service to auth proxy
+- Authentication, keep as is port 4180 is only entry and it goes through the auth proxy
+
+### Current manifest
+
+```yaml
+spec:
+  culling:
+    idleSecondsThreshold: 0
+    maxAgeSecondsThreshold: 0
+  auth:
+    oidc:
+      authorizedEmails:
+      - <user-email>
+      authorizedGroups: []
+      clientId: OidcClientName
+      clientSecret:
+        value: OidcSecretValue
+      enabled: true
+      issuerUrl: https://dev.renku.ch/auth/realms/Renku
+    token: ""
+  jupyterServer:
+    defaultUrl:
+    image:
+    resources:
+    rootDir: /home/jovyanl/work/<project_name>
+  routing:
+    host: dev.renku.ch
+    ingressAnnotations:
+      kubernetes.io/ingress.class: nginx
+      nginx.ingress.kubernetes.io/proxy-body-size: "0"
+      nginx.ingress.kubernetes.io/proxy-buffer-size: 8k
+      nginx.ingress.kubernetes.io/proxy-request-buffering: "off"
+    path: /sessions/<session_name>
+    tls:
+      enabled: true
+      secretName: dev-renku-ch-tls
+  storage:
+    pvc:
+      enabled:
+      mountPath: /home/jovyanl/work
+      storageClassName:
+    size:
+```
+
+### Proposed revised manifest
+
+```yaml
+spec:
+  type: jupyterlab # NEW
+  culling:
+    idleSecondsThreshold: 0
+    maxAgeSecondsThreshold: 0
+  auth:
+    oidc:
+      authorizedEmails:
+      - <user-email>
+      authorizedGroups: []
+      clientId: OidcClientName
+      clientSecret:
+        value: OidcSecretValue
+      enabled: true
+      issuerUrl: https://dev.renku.ch/auth/realms/Renku
+    token: ""
+  server:
+    defaultUrl:
+    image:
+    resources:
+    rootDir: /home/jovyanl/work/<project_name>
+  routing:
+    host: dev.renku.ch
+    ingressAnnotations:
+      kubernetes.io/ingress.class: nginx
+      nginx.ingress.kubernetes.io/proxy-body-size: "0"
+      nginx.ingress.kubernetes.io/proxy-buffer-size: 8k
+      nginx.ingress.kubernetes.io/proxy-request-buffering: "off"
+    path: /sessions/<session_name>
```

Comment:
This is specific to renku - in Amalthea it should be generic

Reply:
Agreed. The value is just for illustration purposes. In amalthea it can be set to whatever.

Reply:
This was just for illustration purposes. Amalthea does not really enforce a value there.